ethcore/src/miner/pool_client.rs

```diff
@@ -136,11 +138,8 @@ impl<'a, C: 'a> pool::client::Client for PoolClient<'a, C> where
 	}
 
 	fn verify_transaction(&self, tx: UnverifiedTransaction)-> Result<SignedTransaction, transaction::Error> {
-		self.engine.verify_transaction_basic(&tx, &self.best_block_header)?;
 		let tx = tx.verify_unordered()?;
-
-		self.verify_signed(&tx)?;
-
+		self.verify_for_pending_block(&tx, &self.best_block_header)?;
```

Comment:
Ah, unfortunately we can't change the order here. `basic` needs to be performed before `verify_unordered`, otherwise we are risking a DoS attack vector.

Maybe just use `engine.machine()` methods here directly.

Reply:
Ok